examples/app/random_tiles/main.py

```diff
@@ -0,0 +1,39 @@
+from bokeh.plotting import Figure
+from bokeh.io import curdoc
+from bokeh.core.properties import (String, List)
+from bokeh.models.tiles import MercatorTileSource
+from bokeh.tile_providers import STAMEN_TONER
+
+class RandomTileSource(MercatorTileSource):
+    __implementation__ = """
+    _ = require "underscore"
+    Util = require "util/util"
+    MercatorTileSource = require "renderer/tile/mercator_tile_source"
+    class RandomTileSource extends MercatorTileSource
+      type: "RandomTileSource"
+      get_image_url: (x, y, z) ->
+        urls = @get('urls')
+        url = urls[Math.floor(Math.random() * urls.length)]
+        image_url = @string_lookup_replace(url, @get('extra_url_vars'))
+        [x, y, z] = @tms_to_wmts(x, y, z)
+        return image_url.replace("{X}", x).replace('{Y}', y).replace("{Z}", z)
+    module.exports =
+      Model: RandomTileSource
+    """
+    urls = List(String)
```

Comment:
You mean in the context of a directory app? I'm not quite sure how that works in a standalone script?